actors/evm/src/interpreter/precompiles.rs

```diff
@@ -93,7 +103,75 @@ fn read_right_pad<'a>(input: impl Into<Cow<'a, [u8]>>, len: usize) -> Cow<'a, [u
     input
 }
 
-// https://github.com/ethereum/go-ethereum/blob/25b35c97289a8db4753cdf5ab7f2b306ec71794d/core/vm/contracts.go#L165
+fn todo<RT: Runtime>(_: &RT, _: &[u8]) -> PrecompileResult {
+    todo!()
+}
+
+fn get_actor_code_cid<RT: Runtime>(rt: &RT, input: &[u8]) -> PrecompileResult {
+    let id = EthAddress(input[..20].try_into().unwrap()).as_id();
+    if id.is_none() {
+        return Err(PrecompileError::InvalidInput);
+    }
+    Ok(rt.get_actor_code_cid(&id.unwrap()).unwrap_or_default().to_bytes())
+}
+
+fn get_randomness<RT: Runtime>(rt: &RT, input: &[u8]) -> PrecompileResult {
+    // 1 + 7 (reserved) + 8 + 8 + 8 = 32
+    let word = read_right_pad(input, 32);
+
+    #[derive(num_derive::FromPrimitive)]
+    #[repr(u8)]
+    enum RandomnessType {
+        Chain = 0,
+        Beacon = 1,
+    }
+
+    let randomness_type = RandomnessType::from_u8(word[0]);
+    // 7 bytes reserved
```

Comment:
Check that they're zero?

Reply:
?

Reply:
s in, check that the reserved bytes aren't set. We can also just ignore them, but it's a _bit_ safer/future proof.

Reply:
ah, good point.